pkg/cloudinit/cloudinit.go

```diff
@@ -0,0 +1,68 @@
+package cloudinit
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	cloudinitv1 "github.com/harvester/node-manager/pkg/apis/node.harvesterhci.io/v1beta1"
+)
+
+const AnnotationHash = "node.harvesterhci.io/cloudinit-hash"
+
+var Directory = "/host/oem"
+
+// RequireLocal ensures that the Elemental cloud-init file described by
+// the given `cloudinit` object is an exact copy of the `cloudinit` object's
+// contents.
+func RequireLocal(cloudinit *cloudinitv1.CloudInit) error {
+	absPath := filepath.Join(Directory, cloudinit.Spec.Filename)
```

Comment:
Use webhook or `while-list` to control the allowed file will be more secure now.

Reply:
Webhook has since merged to the repo which helps secure this